src/axom/sina/core/ID.hpp

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef SINA_ID_HPP
+#define SINA_ID_HPP
+
+/*!
+ ******************************************************************************
+ *
+ * \file ID.hpp
+ *
+ * \brief   Implementation file for Sina ID class
+ *
+ * The Sina schema allows records to have either a local ID or a global ID.
+ * When a global ID is specified, that will be used in the database. When a
+ * local ID is specified, an ID will be automatically generated when inserting
+ * the record into the database.
+ *
+ ******************************************************************************
+ */
+
+#include <string>
+
+#include "conduit.hpp"
+
+namespace axom
+{
+namespace sina
+{
+
+/**
+ * Represents whether an ID is local or global.
+ */
+enum class IDType
+{
+  Local,
+  Global
+};
+
+/**
+ * \brief The ID of a Record
+ *
+ * An ID is used to represent the ID of a Record. This class holds both the
+ * actual ID and whether it is a local or global ID.
+ */
```

Comment:
It's not clear to me from the docs and code in this file what comprises a valid local or global ID, other than it being a string. Are there restrictions that could/should be added here?

Specifically -- is there a uniqueness constraint among ID holders? If so, how is it checked/verified?

Reply:
@HaluskaR I think you might be able to answer this one better than I can

Reply:
Yes, ID is meant to be unique! It serves as an identifier that persists whether the record is in a database, filesystem, has been postprocessed, etc.--often something like `quartz-node22-6.22.24-F64A23`. 

"Global ID" can be considered the "default" behavior.

We ran into some cases where generating unique IDs would cause some complications on the side of the data producer (though whether that's still relevant is something I'd need to look into), and introduced "Local ID" to cover for that. Those are only expected to be unique within the scope of one file, and are replaced by Sina on ingest. 

Speaking of, for verification: this is only checked when using Sina Python components, and only those that agglomerate data. That's intentional, since the uniqueness only really matters when you need to differentiate between records. I'd call it good practice to use things with a low chance of collision though, you never know what'll come up.